packages/modules/solaredge/device.py

```diff
@@ -85,38 +85,21 @@ def add_component(self,
             )
 
     def _update_regs(self, synergy_units: int) -> None:
-        """https://www.solaredge.com/sites/default/files/sunspec-implementation-technical-note.pdf:
-        Only enabled meters are readable, i.e. if meter 1 and 3 are enabled, they are readable as 1st meter and 2nd
-        meter (and the 3rd meter isn't readable). The meter type can be read from the Common block Options field (the
-        same strings that we use in the menus).
-        """
-        def set_regs(meter_id, internal_meter_id):
-            for component in self.components.values():
-                if isinstance(component, SolaredgeExternalInverter) or isinstance(component, SolaredgeCounter):
-                    if component.component_config.configuration.meter_id == meter_id:
-                        log.debug(component.component_config.name+": Interne Meter ID " +
-                                  str(internal_meter_id)+", Synergy Units "+str(synergy_units))
-                        component.regs = MeterRegisters(internal_meter_id, synergy_units)
-
-        meters_in_use = [False]*3
+        meters = [None]*3  # type: List[Union[SolaredgeExternalInverter, SolaredgeCounter, None]]
         for component in self.components.values():
-            if isinstance(component, SolaredgeExternalInverter) or isinstance(component, SolaredgeCounter):
-                meters_in_use[component.component_config.configuration.meter_id-1] = True
-        if meters_in_use[0] is True:
-            set_regs(1, 1)
-            if meters_in_use[1] is True:
-                set_regs(2, 2)
-                if meters_in_use[2] is True:
-                    set_regs(3, 3)
-            else:
-                if meters_in_use[2] is True:
-                    set_regs(3, 2)
-        elif meters_in_use[1] is True:
-            set_regs(2, 1)
-            if meters_in_use[2] is True:
-                set_regs(3, 2)
-        else:
-            set_regs(3, 1)
+            if isinstance(component, (SolaredgeExternalInverter, SolaredgeCounter)):
+                meters[component.component_config.configuration.meter_id-1] = component
+
+        # https://www.solaredge.com/sites/default/files/sunspec-implementation-technical-note.pdf:
+        # Only enabled meters are readable, i.e. if meter 1 and 3 are enabled, they are readable as 1st meter and 2nd
+        # meter (and the 3rd meter isn't readable).
+        meters = list(filter(None, meters))
+        for meter_id, meter in enumerate(meters, start=1):
```

Comment:
`filter(None, meters)` erzeugt einen Generator, der "on demand" die Liste `meters` durchgeht und dabei alle Einträge die nicht "truthy" sind überspringt. Das `list` zwingt Python dann dazu alle Elemente abzurufen und einzeln im Speicher in einer Liste abzulegen. Nicht nötig. `meters = filter(None, meters)` tut alles was benötigt ist.

Ich würde dem ganzen noch entweder einen anderen Namen geben oder inlinen. Also so:
```python
used_meters = filter(None, meters)
```
Oder so:
```python
for meter_id, meter in enumerate(filter(None, meters), start=1):
```
Das reduziert das Risiko, dass man übersieht, dass die Variable nochmal überschrieben wird.